File: src/test/java/ukalus/path/astar/FixedCost.java
/*
 * Created on Feb 10, 2005
 *
 */
package ukalus.path.astar;

/**
 * @author gremopm
 * 
 */
public class FixedCost implements Cost {

  private int cost;

  public FixedCost(int cost) {
    this.cost = cost;
  }

  public double calculate(Node previous, Node current) {
    return cost;
  }

}

File: src/test/kotlin/ukalus/path/astar/FixedCost.kt
/*
 * Created on Feb 10, 2005
 *
 */
package ukalus.path.astar

/**
 * @author gremopm
 */
fun fixedCost(cost: Int): (Node, Node) -> Double {
    return { _: Node, _: Node -> cost.toDouble() }
}